nussl/core/augmentation_utils.py

```diff
@@ -0,0 +1,72 @@
+import tempfile
+from .audio_signal import AudioSignal
+import ffmpeg
+
+def save_audio_signal_to_tempfile(audio_signal, extension=".wav"):
```

Comment:
We should use a different style for temporary files. See here for an example:

https://github.com/nussl/nussl/blob/master/nussl/core/play_utils.py#L147

Essentially, use a context manager to make sure everything is safe. This solution is also portable with Windows.